clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
@hachikuji Alternative, in the case you don't have time to submit that patch or discuss the right behavior, can we first commit this patch so that messages before the bad message will not get lost?